tests/socket.rs

```diff
@@ -554,6 +557,24 @@ fn unix_sockets_supported() -> bool {
     true
 }
 
+// An extension on cygwin. It disables initial handshake on AF_UNIX sockets.
+// https://cygwin.com/git/?p=newlib-cygwin.git;a=commit;h=697b9afe0
+#[cfg(target_os = "cygwin")]
+fn set_no_peercred(s: &Socket) {
```

Comment:
Why is this needed?

Reply:
Well I have considered about that but the maintainers warned me not to do this, because handshake is necessary in most times. In addition, they think that the cygwin's behavior is conformative. I suggest keeping it simple: either add this workaround in the test, or just ignore the test. Most users doesn't connect and accept in the same thread, and if they really do that, they usually use non-blocking mode.

Reply:
Or, maybe we can add a new API to disable peercred, only enabled on cygwin?